File: frontend/src/model/playback-state.ts
export type PlayingState = 'stopped' | 'playing';

export interface PlaybackState {
  playing: PlayingState;
  songId: string;
  sectionId: string;
  queuedSongId: string;
  queuedSectionId: string;
  loopCount: number;
  looping: boolean;
}

File: core/src/model/playback_state.rs
use super::{PlaybackState, PlayingState};

impl PlaybackState {
    pub fn is_playing(&self) -> bool {
        self.playing.enum_value_or_default() == PlayingState::PLAYING
    }

    pub fn is_stopped(&self) -> bool {
        self.playing.enum_value_or_default() == PlayingState::STOPPED
    }
}
